codalab/rest/bundle.py

```diff
@@ -1,19 +1,517 @@
+            if spec.type == list or not spec.generated:
+                metadata_dict[spec.key] = spec.get_constructor()()
+            metadata_spec_dict[spec.key] = spec
+        for row in rows:
+            (maybe_unicode_key, value) = (row['metadata_key'], row['metadata_value'])
+            # If the key is Unicode text (which is the case if it was extracted from a
+            # database), cast it to a string. This operation encodes it with UTF-8.
+            key = str(maybe_unicode_key)
+            if key not in metadata_spec_dict:
+                continue  # Somewhat dangerous since we might lose information
+
+            spec = metadata_spec_dict[key]
+            if spec.type == list:
+                metadata_dict[key].append(value)
+            else:
+                if metadata_dict.get(key):
+                    # Should be internal error
+                    from codalab.common import UsageError
+                    raise UsageError(
+                        'Got duplicate values %s and %s for key %s' %
+                        (metadata_dict[key], value, key)
+                    )
+                # Convert string to the right type (e.g., string to int)
+                metadata_dict[key] = spec.get_constructor()(value)
+        return metadata_dict
+
+    def _deserialize(self, metadata_dict, attr, bundle_info):
+        """
+        Serialize this metadata object and return a list of dicts that can be
+        saved to a MySQL table.
+        Originally Metadata.to_dicts and Metadata.validate
+        """
+        metadata_specs = self._get_metadata_specs(bundle_info)
+        expected_keys = set(spec.key for spec in metadata_specs)
+        for key in metadata_dict:
+            if key not in expected_keys:
+                raise ValidationError('Unexpected metadata key: %s' % (key,))
+        result = []
+        for spec in metadata_specs:
+            if spec.key in metadata_dict:
+                raw_value = metadata_dict[spec.key]
+                if raw_value is None:
+                    continue
+                if spec.validate is not None:
+                    spec.validate(raw_value)
+                if spec.type == float and isinstance(raw_value, int):
+                    # cast int to float
+                    raw_value = float(raw_value)
+                if raw_value is not None and not isinstance(raw_value, spec.type):
+                    raise ValidationError(
+                        'Metadata value for %s should be of type %s, was %s (type %s)' %
+                        (spec.key, spec.type, raw_value, type(raw_value))
+                    )
+                values = raw_value if spec.type == list else (raw_value,)
+                for value in values:
+                    result.append({
+                        'metadata_key': unicode(spec.key),
+                        'metadata_value': unicode(value),
+                    })
+            elif not spec.generated:
```

Comment:
Seems weird that you have to specify everything if you only want to update 1 field.